ironic_prometheus_exporter/parsers/ipmi.py

```diff
@@ -0,0 +1,166 @@
+    metric ('Temp (0x1)' and 'Temp (0x2)') and one entry for 'Inlet Temp (0x5)'
+    and other for 'Exhaust Temp (0x6)', this will produce a dictionary where
+    the keys are the entries and the values are the respective label to be used
+    when writting the metrics in the Prometheus format.
+    {'Inlet Temp (0x5)': '{node_name=...}',
+     'Exhaust Temp (0x6)': '{node_name=...}',
+     'Temp (0x1)': '{node_name=...,sensor=Temp1}',
+     'Temp (0x2)': '{node_name=...,sensor=Temp2}'}
+
+    returns: a dictionarty of dictionaries {<entry>: {label_name: label_value}}
+    """
+    LOG.info('extract_labels function called with: entries=%s | payload=%s | \
+             node_name=%s' % (str(entries), str(payload), node_name))
+    if len(entries) == 1:
+        labels = {'node_name': node_name,
+                  'entity_id': payload[entries[0]]['Entity ID']}
+        return {entries[0]: labels}
+    entries_labels = {}
+    for entry in entries:
+        try:
+            entity_id = payload[entry]['Entity ID']
+            sensor_id = payload[entry]['Sensor ID']
+            metric_label = {'node_name': node_name,
+                            'entity_id': entity_id,
+                            'sensor_id': sensor_id}
+            entries_labels[entry] = metric_label
+        except Exception as e:
+            LOG.exception(e)
+    return entries_labels
+
+
+def extract_values(entries, payload, use_ipmi_format=True):
+    LOG.info('extract_values function called with: entries=%s | payload=%s |'
+             % (str(entries), str(payload)))
+    values = {}
+    for entry in entries:
+        try:
+            no_values = ['No Reading', 'Disabled']
+            if payload[entry]['Sensor Reading'] in no_values:
+                values[entry] = None
+            else:
+                sensor_values = payload[entry]['Sensor Reading'].split()
+                if not use_ipmi_format:
+                    if not re.search(r'(\d+(\.\d*)?|\.\d+)', sensor_values[0]):
+                        raise Exception("No valid value in Sensor Reading")
+                    values[entry] = sensor_values[0]
+                if len(sensor_values) > 1:
+                    values[entry] = sensor_values[0]
+                elif sensor_values[0] == "0h":
+                    values[entry] = 0
+                else:
+                    values[entry] = 1
+        except Exception as e:
+            LOG.exception(e)
+    return values
+
+
+def prometheus_format(payload, node_name, ipmi_metric_registry,
+                      available_metrics, use_ipmi_format):
+    for metric in available_metrics:
+        entries = available_metrics[metric]
+        labels = extract_labels(entries, payload, node_name)
+        values = extract_values(entries, payload,
+                                use_ipmi_format=use_ipmi_format)
+        if all(v is None for v in values.values()):
+            continue
+        g = Gauge(metric, '', labelnames=labels.get(entries[0]).keys(),
```

Comment:
nit: `keys()` is an iterator in python 3 and is usually redundant with dicts. so probably something like `labelnames=list(labels.get(entries[0]))`